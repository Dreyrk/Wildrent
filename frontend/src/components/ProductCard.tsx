import Link from 'next/link';
import React from 'react';

type ProductCardProps = {
  name: string;
  description: string;
  image: string;
  priceHT: number;
  link: string;
};
function ProductCard({ name, description, image, priceHT, link }: ProductCardProps) {
  return (
    <div>
      <Link href={link}>
        <h3> {name}</h3>
        <img src={image} alt={name} />
        <p>{description}</p>
        <p>{priceHT}â‚¬</p>
      </Link>
    </div>
  );
}

export default ProductCard;
