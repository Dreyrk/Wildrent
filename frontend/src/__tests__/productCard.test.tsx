import ProductCard from '@/components/ProductCard';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';

describe('ProductCard Component', () => {
  it('renders correctly', () => {
    const view = render(
      <ProductCard
        name="Guirlande"
        image="https://img.com/i.png"
        description="Cupcake ipsum dolor sit amet tootsie roll pastry."
        priceHT={1}
        link="/products/1"
      ></ProductCard>,
    );
    const link = screen.getByRole('link');
    const heading = screen.getByRole('heading', { level: 3 });
    expect(link).toHaveAttribute('href', '/products/1');
    expect(heading).toBeInTheDocument();
    expect(screen.getByText('Guirlande')).toBeInTheDocument();
    expect(view.baseElement).toMatchSnapshot();
  });
});
