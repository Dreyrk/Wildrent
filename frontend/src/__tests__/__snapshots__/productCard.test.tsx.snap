// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard Component renders correctly 1`] = `
<body>
  <div>
    <div>
      <a
        href="/products/1"
      >
        <h3>
           
          Guirlande
        </h3>
        <img
          alt="Guirlande"
          src="https://img.com/i.png"
        />
        <p>
          Cupcake ipsum dolor sit amet tootsie roll pastry.
        </p>
        <p>
          1
          â‚¬
        </p>
      </a>
    </div>
  </div>
</body>
`;
